src/integer_mod_q/modulus_polynomial_ring_zq/get.rs

```diff
@@ -39,6 +42,74 @@ impl ModulusPolynomialRingZq {
         }
         out
     }
+
+    /// Returns the degree of a polynomial [`ModulusPolynomialRingZq`] as a [`i64`].
+    /// The zero polynomial has degree '-1'.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::ModulusPolynomialRingZq;
+    /// use std::str::FromStr;
+    ///
+    /// let poly = ModulusPolynomialRingZq::from_str("4  0 1 2 3 mod 7").unwrap();
+    ///
+    /// let degree = poly.get_degree(); // This would only return 3
+    /// ```
+    pub fn get_degree(&self) -> i64 {
+        unsafe { fq_ctx_degree(self.get_fq_ctx_struct()) }
+    }
+}
+
+#[cfg(test)]
+mod test_get_degree {
+    use crate::integer_mod_q::ModulusPolynomialRingZq;
+    use std::str::FromStr;
+
+    /// Ensures that degree is working for constant polynomials.
+    #[test]
+    fn degree_constant() {
+        let degrees = [0, 1, 3, 7, 15, 32, 120];
+        for degree in degrees {
+            let modulus_ring = ModulusPolynomialRingZq::from_str(&format!(
+                "{}  {}1 mod 17",
+                degree + 1,
+                "0 ".repeat(degree)
+            ))
+            .unwrap();
```

Comment:
better instantiate a PolyOverZq with mod 17 and set the degree coefficient to be 1, looks much cleaner

then instantiate a ModulusPolynomialRingZq from a PolyOverZq